src/Aspire.Hosting.Azure.Functions/AzureFunctionsProjectResourceExtensions.cs

```diff
@@ -50,9 +51,15 @@ public static class AzureFunctionsProjectResourceExtensions
             {
                 var configureInfrastructure = (AzureResourceInfrastructure infrastructure) =>
                 {
+
+                    var principalTypeParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalType, typeof(string));
+                    var principalIdParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalId, typeof(string));
+                    infrastructure.Add(principalTypeParameter);
+                    infrastructure.Add(principalIdParameter);
```

Comment:
This is unfortunate because it results in duplicate parameters in the bicep. @tg-msft - is that something we want to protect against in Azure.Provisioning? Adding duplicate parameters to an `Infrastructure` object?

Maybe we shouldn't be adding these parameters to the infastructure expicitly, and letting our parameter fix up code take care of it, like we were before.

Reply:
The fixup works well if there's no special handling (e.g secrets or non string types)

Reply:
What if we had a `GetOrAddParameter(string name, Type? type = null)` method on `Infrastructure`?

Reply:
These would all be:

```C#
var principalTypeParameter = infrastructure.GetOrAddParameter(AzureBicepResource.KnownParameters.PrincipalType);
var principalIdParameter = infrastructure.GetOrAddParameter(AzureBicepResource.KnownParameters.PrincipalId);
```